pkg/bytecode/code.go

```diff
@@ -83,6 +83,14 @@ const (
 	// OpIndex represents an index operator used on an array, map or
 	// string variable.
 	OpIndex
+	// OpSlice represents a slice operator used on an array or string. The top
+	// three elements are read from the stack and are the end index, start index
+	// and the data structure being operated on. OpNull is used as a default
+	// when the start or end are not specified by the user program.
+	OpSlice
+	// OpNull represents a null value. It is used as a sensible default for
+	// the start or end of an OpSlice.
+	OpNull
```

Comment:
I don't think this should be called `OpNull`. Evy does not have null values so I think this can be confusing down the track. I would perhaps call it `OpNoneVal` just like the type and document it as:
> OpNoneVal represents an unspecified value used where values are optional and unspecified such as
> the start and end value of a slice index or the values in a step range.

And then update any reference to "null" values in naming to refer to "none" or similar.